hpat/tests/gen_test_data.py

```diff
@@ -5,102 +5,132 @@
 import pandas as pd
 
 
+class ParquetGenerator:
+    GEN_KDE_PQ_CALLED = False
+    GEN_PQ_TEST_CALLED = False
+
+    @classmethod
+    def gen_kde_pq(cls, file_name='kde.parquet', N=101):
+        if not cls.GEN_KDE_PQ_CALLED:
+            df = pd.DataFrame({'points': np.random.random(N)})
+            table = pa.Table.from_pandas(df)
+            row_group_size = 128
+            pq.write_table(table, file_name, row_group_size)
+            cls.GEN_KDE_PQ_CALLED = True
+
+    @classmethod
+    def gen_pq_test(cls):
+        if not cls.GEN_PQ_TEST_CALLED:
+            df = pd.DataFrame(
+                {
+                    'one': [-1, np.nan, 2.5, 3., 4., 6., 10.0],
+                    'two': ['foo', 'bar', 'baz', 'foo', 'bar', 'baz', 'foo'],
+                    'three': [True, False, True, True, True, False, False],
+                    # float without NA
+                    'four': [-1, 5.1, 2.5, 3., 4., 6., 11.0],
+                    # str with NA
+                    'five': ['foo', 'bar', 'baz', None, 'bar', 'baz', 'foo'],
+                }
+            )
+            table = pa.Table.from_pandas(df)
+            pq.write_table(table, 'example.parquet')
+            pq.write_table(table, 'example2.parquet', row_group_size=2)
+            cls.GEN_PQ_TEST_CALLED = True
+
+
+class SparkGenerator:
+
+    @staticmethod
+    def generate():
+        from pyspark.sql import SparkSession
+        from pyspark.sql.types import (
+            StructType, StructField, DateType, TimestampType)
+
+        # test datetime64, spark dates
+        dt1 = pd.DatetimeIndex(['2017-03-03 03:23',
+                                '1990-10-23', '1993-07-02 10:33:01'])
+        df = pd.DataFrame({'DT64': dt1, 'DATE': dt1.copy()})
+        df.to_parquet('pandas_dt.pq')
+
+        spark = SparkSession.builder.appName("GenSparkData").getOrCreate()
+        schema = StructType([StructField('DT64', DateType(), True),
+                             StructField('DATE', TimestampType(), True)])
+        sdf = spark.createDataFrame(df, schema)
+        sdf.write.parquet('sdf_dt.pq', 'overwrite')
+
+        spark.stop()
+
+
 def gen_lr(file_name, N, D):
-    points = np.random.random((N,D))
+    points = np.random.random((N, D))
```

Comment:
Please, next time do not mix code style changes and algorithm changes.

Reply:
OK